src/Components/CatProfile.tsx

```diff
@@ -1,38 +1,138 @@
+					setCatPicture(profile.pictureurl);
+					setCatDescription(profile.description);
+				}
+				setIsLoading(false);
+			}
+		);
+	}, []);
+
+	if (isLoading) {
+		return <p>Loading...</p>;
+	}
+
+	const handleProfileEditBy = (event: ChangeEvent<HTMLInputElement>) => {
+		//Handle editing process of Cat's profile, allow users to edit the profile
+		const { name, value } = event.target;
+
+		if (name === 'name') {
+			setCatName(value);
+		} else if (name === 'pictureurl') {
+			setCatPicture(value);
+		} else if (name === 'description') {
+			setCatDescription(value);
+		}
+	};
+
+	const handleSavedChanges = () => {
+		const updatedProfile = {
+			id: catProfiles[0]?.id,
+			name: catName,
+			pictureurl: catPicture,
+			description: catDescription,
+			device_id: catProfiles[0].device_id,
+			owner_id: catProfiles[0].owner_id,
+			updated_at: new Date().toISOString(),
+			created_at: catProfiles[0].created_at,
+			deleted_at: null,
+		};
+		updateCatProfile(updatedProfile)
+			.then(() => {
+				setCatProfiles((previousProfiles) =>
+					previousProfiles?.map((profile) =>
+						profile.id === updatedProfile.id ? updatedProfile : profile
+					)
+				);
+				setMessage('Updates have been saved');
+			})
+			.catch(() => {
+				setMessage('Updates havent been saved');
+			});
+	};
+
+	const handleDeleteProfile = () => {
+		deleteCatProfile(REACT_APP_LOGGED_IN_USER as string)
+			.then(() => {
+				setCatProfiles((fetchedProfiles) =>
+					fetchedProfiles.filter((profile) => profile.id !== profile.id)
+				);
+				setMessage('Profile Removed');
+			})
+			.catch(() => {
+				setMessage('Failed to delete the profile');
+			});
+	};
+  
+  return (
+		<div className="w-72 p-20 m-auto bg-emerald-100	">
+			<h2 className="text-2xl font-bold text-red-950	mb-20">
+				Cat&apos;s Profile
+			</h2>
+			{message && (
+				<div
+					className={`text-2xl font-bold  mb-20
+					${message === 'Updates have been saved' ? 'text-red-950' : ''}
+					${message === 'Updates havent been saved' ? 'text-red-950' : ''}
+					${message === 'Profile Removed' ? 'text-red-950' : ''}
+					${message === 'Failed to delete the profile' ? 'text-red-950' : ''}`}
+				>
+					{message}
+				</div>
+			)}
 			<div className="flex flex-col">
 				<FormInput
 					label="Name"
 					type="text"
 					name="name"
-					value="Cat name"
-					onChange={() => {}}
+					value={catName}
+					onChange={handleProfileEditBy}
 				/>
 				<FormInput
-					label="Pricture"
+					label="Picture"
 					type="text"
 					name="pictureurl"
-					value="Picture URL"
-					onChange={() => {}}
+					value={catPicture}
+					onChange={handleProfileEditBy}
 				/>
 				<FormInput
 					label="Description"
-					type=""
+					type="text"
 					name="description"
-					value="A good girl"
-					onChange={() => {}}
+					value={catDescription}
+					onChange={handleProfileEditBy}
 				/>
-				<Button onClick={() => {}}>Save</Button>
+				<Button onClick={handleSavedChanges}>Save</Button>
```

Comment:
The button that should be triggered by the return key should be marked as 'submit'